app/src/main/java/com/android/sample/model/preferences/PreferencesRepository.kt

```diff
@@ -0,0 +1,12 @@
+package com.android.sample.model.preferences
+
+interface PreferencesRepository {
+
+  fun init(onSuccess: () -> Unit)
+
+  fun getPreferences(onSuccess: (Preferences) -> Unit, onFailure: (Exception) -> Unit)
+
+  fun updatePreferences(pref: Preferences, onSuccess: () -> Unit, onFailure: (Exception) -> Unit)
+
+  fun deletePreferences(onSuccess: () -> Unit, onFailure: (Exception) -> Unit)
+}
```

Comment:
Can you add the Kdoc so that its clearer for the team what each function does ?

Reply:
Done.